docs/next/manual/msp/guides/apigw/mse_policy.md

```diff
@@ -0,0 +1,302 @@
+
+### 配置说明
+
+| 配置项                 | 含义                                                         |
+| :--------------------- | :----------------------------------------------------------- |
+| 鉴别用户的 Cookie 名称 | 用于生成 CSRF Token，未携带该 Cookie 则不会进行 CSRF 校验，后端识别该 Cookie 过期时需配合前端进行清除，避免缺失 CSRF Token 导致登录报错 |
+| 关闭校验的 HTTP 方法 | 对于这类请求方法，将跳过 CSRF 校验，但仍会种下 CSRF Token 的 Cookie |
+| Token 名称       | 网关将生成的 CSRF Token 设置在该名称的 Cookie 里，前端需从 Cookie 中获取 CSRF Token，带在同名请求头里发起请求 |
+| Token Cookie 的生效域名 | 如未填写，则仅针对发起请求的域名种下 CSRF Token 的 Cookie |
+| Cookie 开启 Secure 属性 | 开启 Secure 后，所有 HTTP 请求均无法通过 CSRF 校验    |
+| Token 过期时间     | 过期 Token 将无法通过校验                                |
+| 校验失败的状态码   | CSRF 校验失败时返回的 HTTP 状态码                      |
+| 校验失败的应答     | CSRF 校验失败时返回的 HTTP 应答                        |
+
+
+### 使用说明
+
+由于 MSE 网关 erda-csrf 插件是自定义开发，受限于框架 tinygo 的限制，在 MSE 网关 跨域防护 csrf 网关路由策略的情况下，用户的应用场景应当满足如下图所示的情况，核心要点是:
+* 步骤 1（用户登录用户中心): 用户登录页的原始请求不带 鉴别用户的 Cookie ，登录页的请求返回中需要在 "Set-Cookie" 中设置 鉴别用户的 Cookie 名称 （如 `uc-token`) 及其对应的值;
+* 步骤 2（MSE 网关，用户无需关心): MSE 网关 erda-csrf 插件根据登录页的返回，从返回的 "Set-Cookie" 中获取 鉴别用户的 Cookie  （如 `uc-token`)（ 如果获取不到，则默认设置 `uc-token` = `1ba855fcb59caa9d82370495d`），然后采用相关算法生成 csrf token, 将其作为 Token 名称 （如 `csrf-token`） 对应的值设置到返回的  "Set-Cookie" 中
+* 步骤 3（用户访问业务详情）: 用户根据登录页返回的  "Set-Cookie" 中的 鉴别用户的 Cookie （如 `uc-token`)  和 Token 名称（如 `csrf-token`） 发起下一步请求。（如果用户的登录页返回的逻辑中不设置 鉴别用户的 Cookie （如 `uc-token`)，则默认设置 鉴别用户 Cookie  （如 `uc-token`) 为 `uc-token` = `1ba855fcb59caa9d82370495d`））
+
+
+![](https://terminus-paas.oss-cn-hangzhou.aliyuncs.com/paas-doc/2019/08/30/bb252336-b2cf-44d4-89be-92581ccd2d52.jpg)
+
+
+**注意**，登录用户中心的服务 和 后续请求的服务需要用户自己开发且满足请求返回的 header 和/或 cookie 设置，用户应当评估开启策略后 进行 token 生成和 token 校验 造成的开销。
+
+#### 使用示例
+
+假设登录页返回的  "Set-Cookie" 中的  鉴别用户 Cookie （如 `uc-token`) 为  和 Token 名称（如 `csrf-token`）分别为:
+* uc-token=`1ba855fcb59caa9d82370495d`
+* csrf-token: `eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJhdWQiOiJ1cHN0cmVhbSIsImV4cCI6IjE2ODQ4MzU1ODgiLCJpYXQiOiIxNjg0ODMzNzg4IiwiaXNzIjoiRXJkYS1BUEktR2F0ZXdheSIsImp0aSI6IjFiYTg1NWZjYjU5Y2FhOWQ4MjM3MDQ5NWQiLCJuYmYiOiIxNjg0ODMzNzg4Iiwic3ViIjoiZnJvbnRlbmQifQ.afZWTG5s0b7l_-9e1-iYEe8W5xMEIC1kRgCSe607DZs`
+
+则使用  鉴别用户 Cookie 和 Token 名称 分别在 Cookie 和 Header 中设置相关配置发起请求即可：
+* uc-token 设置在 Cookie 中
+* csrf-token 设置在 Header 中
+
+##### 正常返回:
+
+```bash
+curl -v 'http://test-gateway.mse-daily.terminus.io/testmse/test/policy' \
+    -H 'Cookie: uc-token=1ba855fcb59caa9d82370495d; OPENAPISESSION=937ec28c-b3a7-4f64-95ad-4fa0e4e19ad3' \
+    -H 'Csrf-Token: eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJhdWQiOiJ1cHN0cmVhbSIsImV4cCI6IjE2ODQ4MzU1ODgiLCJpYXQiOiIxNjg0ODMzNzg4IiwiaXNzIjoiRXJkYS1BUEktR2F0ZXdheSIsImp0aSI6IjFiYTg1NWZjYjU5Y2FhOWQ4MjM3MDQ5NWQiLCJuYmYiOiIxNjg0ODMzNzg4Iiwic3ViIjoiZnJvbnRlbmQifQ.afZWTG5s0b7l_-9e1-iYEe8W5xMEIC1kRgCSe607DZs' \
+    --compressed \
+    --insecure
+*   Trying 116.62.243.103:80...
```

Comment:
Sensitive information needs to be desensitized.

Reply:
Masked.